test/jit/test_with.py

```diff
@@ -548,3 +548,18 @@ def test_exit_incorrect_types(x, c):
             self.checkScript(
                 test_exit_incorrect_types, (test_tensor, ExitIncorrectTypes())
             )
+
+    def test_with_no_grad(self):
```

Comment:
The only part of this that tests `no_grad` is `requires_grad`. Are there any numeric tests I can run here?

Reply:
see below, you should look into what we test with no_grad when not scripting.